src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,165 @@
+     * @throws NoArgumentException If only the command is given without any arguments.
+     */
+    public boolean parseInput(String userInput, boolean fromSave)
+            throws WrongArgumentException, FileParseException, NoArgumentException {
+        if (isListCommand(userInput.split(" ")[0])) {
+            this.parseListCommands(userInput, fromSave);
+        } else if (userInput.equals("bye")) {
+            Ui.showBye();
+            return true;
+        } else if (userInput.equals("list")) {
+            Ui.displayList(list);
+        } else if (userInput.equals("/?")) {
+            Ui.showHelp();
+        } else {
+            System.out.println("what's this?! REDO!!!!");
+        }
+        return false;
+    }
+
+    private boolean isListCommand(String input) {
+        for (ListCommands c : ListCommands.values()) {
+            if (c.name().equals(input)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Parses the string if the command is one that is handled by the TaskList, and it will call the appropriate methods
+     *
+     * @param input String from save file or user.
+     * @param fromSave true if string is from file, false if not
+     * @throws WrongArgumentException If argument is of the wrong format.
+     * @throws FileParseException If there's an error when the line was written to save file previously.
+     * @throws NoArgumentException If only the command is given without any arguments.
+     */
+    private void parseListCommands(String input, boolean fromSave)
+            throws WrongArgumentException, FileParseException, NoArgumentException {
+        String mark = null;
+        String[] arr;
+        if (fromSave) {
+            String[] item = input.split("\\|");
+            mark = item[1];
+            arr = item[0].split(" ", 2);
+        } else {
+            arr = input.split(" ", 2);
+        }
+        ListCommands command = ListCommands.valueOf(arr[0]);
+
+        try {
+            switch (command) {
+            case mark: {
+                int index = Integer.parseInt(arr[1]) - 1;
+                Ui.markTaskDone(list.markAsDone(index));
+                return;
+            }
+            case unmark: {
+                int index = Integer.parseInt(arr[1]) - 1;
+                Ui.markTaskNotDone(list.markAsNotDone(index));
+                return;
+            }
+            case delete: {
+                int index = Integer.parseInt(arr[1]) - 1;
+                Ui.deleteTask(list.delete(index));
+                return;
+            }
+            case find:
+                String keyword = arr[1];
+                Ui.findKeyword(keyword, list.searchFor(keyword));
+                return;
+            }
+        } catch (NumberFormatException | IndexOutOfBoundsException e) {
+            throw new WrongArgumentException(arr[1], e);
+        }
+
+        switch (command) {
```

Comment:
Is there a reason why 2 switch commands is needed? 
Can it be condensed into a single switch statement? 🤔